test/integration_test/migration_stork_failures_test.go

```diff
@@ -0,0 +1,197 @@
+	require.NoError(t, err, "Error scheduling cluster pair")
+
+	// apply migration specs
+	currMigNamespace := migrationAppKey + "-" + nsKey + "-dest"
+	currMig, err := createMigration(t, migrationKey, currMigNamespace, "remoteclusterpair", currMigNamespace, &includeResourcesFlag, &startApplicationsFlag)
+	require.NoError(t, err, "failed to create migration %s in namespace: %s", currMig.Name, currMig.Namespace)
+
+	// Wait for migration to start
+	err = waitForMigrationToStart(currMig.Name, currMig.Namespace, migrationRetryTimeout)
+	require.NoError(t, err, "Migration %s failed to start in namespace: %s", currMig.Name, currMig.Namespace)
+
+	// Change kubeconfig to dest and delete stork pods
+	err = setDestinationKubeConfig()
+	require.NoError(t, err, "failed to set kubeconfig to destination cluster for deleting stork pods.")
+	logrus.Infof("Migration is in progress. Deleting stork pods.")
+	storkPods, err := core.Instance().GetPods(storkNamespace, storkLabel)
+	require.NoError(t, err, "failed to get stork pods.")
+
+	err = core.Instance().DeletePods(storkPods.Items, false)
+	require.NoError(t, err, "failed to delete stork pods.")
+
+	// Change kubeconfig back to source. Ensure migration is successful
+	err = setSourceKubeConfig()
+	require.NoError(t, err, "failed to set kubeconfig to source cluster to check for migration.")
+	err = waitForMigrationToComplete(currMig.Name, currMig.Namespace, migrationRetryTimeout)
+	require.NoError(t, err, "Migration %s failed to start in namespace: %s", currMig.Name, currMig.Namespace)
+
+	// Change kubeconfig to destination. Check stork pods are running, for sanity
+	err = setDestinationKubeConfig()
+	require.NoError(t, err, "failed to set kubeconfig to destination cluster for checking stork pods.")
+	_, err = core.Instance().GetPods(storkNamespace, storkLabel)
+	require.NoError(t, err, "failed to get stork pods after migration.")
+
+	// Change kubeconfig back to source. Clean up
+	err = setSourceKubeConfig()
+	require.NoError(t, err, "failed to set kubeconfig to source cluster before cleanup.")
+	destroyAndWait(t, ctxs)
+	err = deleteAndwaitForMigrationDeletion(currMig.Name, currMig.Namespace, migrationRetryTimeout)
+	require.NoError(t, err, "Migration %s failed to delete in namespace: %s", currMig.Name, currMig.Namespace)
+}
+
+func waitForMigrationToStart(name, namespace string, timeout time.Duration) error {
+	getMigration := func() (interface{}, bool, error) {
+		migration, err := storkops.Instance().GetMigration(name, namespace)
+		if err != nil {
+			return "", false, err
+		}
+
+		if migration.Status.Status != storkv1.MigrationStatusInProgress {
+			return "", true, fmt.Errorf("Migration %s in %s has not started yet.Status: %s. Retrying ", name, namespace, migration.Status.Status)
+		}
+		return "", false, nil
+	}
+	_, err := task.DoRetryWithTimeout(getMigration, timeout, backupWaitInterval)
+	return err
+}
+
+func waitForMigrationToComplete(name, namespace string, timeout time.Duration) error {
```

Comment:
you can use sched-ops `ValidateMigration` api to check migration completion -> https://github.com/portworx/sched-ops/blob/master/k8s/stork/migration.go

Reply:
Fixed.